WeakAuras/GenericTrigger.lua

```diff
@@ -3472,66 +3492,95 @@ function WeakAuras.WatchUnitChange(unit)
           reactionUpdate(unit, eventsToSend)
         end
       elseif event == "NAME_PLATE_UNIT_ADDED" then
+        if not watchUnitChange.trackedUnits[unit] then
+          return
+        end
         unitUpdate(unit, eventsToSend)
-        markerUpdate(unit, eventsToSend)
-        reactionUpdate(unit, eventsToSend)
+        markerInit(unit)
+        reactionInit(unit)
       elseif event == "NAME_PLATE_UNIT_REMOVED" then
+        if not watchUnitChange.trackedUnits[unit] then
+          return
+        end
         unitUpdate(unit, eventsToSend)
-        markerUpdate(unit, eventsToSend)
-        reactionUpdate(unit, eventsToSend)
+        markerClear(unit)
+        reactionClear(unit)
       elseif event == "INSTANCE_ENCOUNTER_ENGAGE_UNIT" then
         for i = 1, 5 do
           local unit = "boss" .. i
-          unitUpdate(unit, eventsToSend)
-          markerUpdate(unit, eventsToSend)
-          reactionUpdate(unit, eventsToSend)
+          if watchUnitChange.trackedUnits[unit] then
+            unitUpdate(unit, eventsToSend)
+            markerInit(unit)
+            reactionInit(unit)
+          end
         end
       elseif event == "PLAYER_TARGET_CHANGED" then
+        if not watchUnitChange.trackedUnits["target"] then
+          return
+        end
         unitUpdate("target", eventsToSend)
-        markerUpdate("target", eventsToSend)
-        reactionUpdate("target", eventsToSend)
+        markerInit("target")
+        reactionInit("target")
       elseif event == "PLAYER_FOCUS_CHANGED" then
+        if not watchUnitChange.trackedUnits["focus"] then
+          return
+        end
         unitUpdate("focus", eventsToSend)
-        markerUpdate("focus", eventsToSend)
-        reactionUpdate("focus", eventsToSend)
+        markerInit("focus")
+        reactionInit("focus")
       elseif event == "PLAYER_SOFT_ENEMY_CHANGED" then
+        if not watchUnitChange.trackedUnits["softenemy"] then
+          return
+        end
         unitUpdate("softenemy", eventsToSend)
-        markerUpdate("softenemy", eventsToSend)
-        reactionUpdate("softenemy", eventsToSend)
+        markerInit("softenemy")
+        reactionInit("softenemy")
       elseif event == "PLAYER_SOFT_FRIEND_CHANGED" then
+        if not watchUnitChange.trackedUnits["softfriend"] then
+          return
+        end
         unitUpdate("softfriend", eventsToSend)
-        markerUpdate("softfriend", eventsToSend)
-        reactionUpdate("softfriend", eventsToSend)
+        markerInit("softfriend")
+        reactionInit("softfriend")
       elseif event == "RAID_TARGET_UPDATE" then
         for unit in pairs(watchUnitChange.raidmark) do
           markerUpdate(unit, eventsToSend)
         end
       elseif event == "UNIT_FACTION" then
-        if unit:sub(1, 9) == "nameplate" then
+        if watchUnitChange.trackedUnits[unit] then
```

Comment:
When the reaction was initially introduced it was only supposed to be only available on the nameplate unit, though it turns out that the option was not disabled/hidden for other unit types in the Unit Characteristics trigger. 

Thus, fix that by removing this check, and making the reaction option available in all triggers.